crypto/threads_win.c

```diff
@@ -9,49 +9,102 @@
 
 #if defined(_WIN32)
 # include <windows.h>
+# if defined(_WIN32_WINNT) && _WIN32_WINNT >= 0x600
+#  include <synchapi.h>
+#  define USE_RWLOCK
+# endif
 #endif
 
 #include <openssl/crypto.h>
 
 #if defined(OPENSSL_THREADS) && !defined(CRYPTO_TDEBUG) && defined(OPENSSL_SYS_WINDOWS)
 
+# ifdef USE_RWLOCK
+typedef enum {
+    SRWLOCK_MODE_SHARED = 0,
+    SRWLOCK_MODE_EXCLUSIVE
+} LOCK_MODE;
+
+typedef struct {
+    SRWLOCK lock;
+    LOCK_MODE mode;
```

Comment:
I'd not bother with an enum here, it's a Boolean value.  Simply `int shared` or `int exclusive` is enough.  Keep it simple.